tests/chains/ethereum/scripts/cmd/helper/chain.go

```diff
@@ -0,0 +1,224 @@
+	"github.com/hyperledger-labs/yui-relayer/core"
+)
+
+type PathConfig struct {
+	Src core.PathEnd `json:"src"`
+	Dst core.PathEnd `json:"dst"`
+}
+
+type ChainConfig struct {
+	Chain  ethereum.ChainConfig `json:"chain"`
+	Prover ProverConfig         `json:"prover"`
+}
+
+type ProverConfig struct {
+	Type string `json:"@type"`
+}
+
+type Chain struct {
+	chainID        int64
+	client         *client.ETHClient
+	mnemonicPhrase string
+	keys           map[uint32]*ecdsa.PrivateKey
+
+	ChainConfig ChainConfig
+
+	// Core Modules
+	IBCHandler    ibchandler.Ibchandler
+	IBCCommitment ibccommitment.Ibccommitmenttesthelper
+
+	// App Modules
+	SimpleToken   simpletoken.Simpletoken
+	ICS20Transfer ics20transferbank.Ics20transferbank
+	ICS20Bank     ics20bank.Ics20bank
+
+	// IBC specific helpers
+	ClientIDs []string // ClientID's used on this chain
+
+	// Channel specific helpers
+	Channel core.PathEnd
+}
+
+func NewChain(pathEnd core.PathEnd, chainConfig ChainConfig, client *client.ETHClient, mnemonicPhrase string, simpleTokenAddress, ics20TransferBankAddress, ics20BankAddress string) *Chain {
+	ibcHandler, err := ibchandler.NewIbchandler(chainConfig.Chain.IBCAddress(), client)
+	if err != nil {
+		log.Print(err)
+		return nil
+	}
+	simpletoken, err := simpletoken.NewSimpletoken(common.HexToAddress(simpleTokenAddress), client)
+	if err != nil {
+		log.Print(err)
+		return nil
+	}
+	ics20transfer, err := ics20transferbank.NewIcs20transferbank(common.HexToAddress(ics20TransferBankAddress), client)
+	if err != nil {
+		log.Print(err)
+		return nil
+	}
+	ics20bank, err := ics20bank.NewIcs20bank(common.HexToAddress(ics20BankAddress), client)
+	if err != nil {
+		log.Print(err)
+		return nil
+	}
+
+	return &Chain{
+		client:         client,
+		chainID:        chainConfig.Chain.EthChainId,
+		ChainConfig:    chainConfig,
+		mnemonicPhrase: mnemonicPhrase,
+		keys:           make(map[uint32]*ecdsa.PrivateKey),
+
+		IBCHandler: *ibcHandler,
+
+		SimpleToken:   *simpletoken,
+		ICS20Transfer: *ics20transfer,
+		ICS20Bank:     *ics20bank,
+		Channel:       pathEnd,
+	}
+}
+
+func (chain *Chain) TxOpts(ctx context.Context, index uint32) *bind.TransactOpts {
+	return makeGenTxOpts(big.NewInt(chain.chainID), chain.prvKey(index))(ctx)
+}
+
+func (chain *Chain) prvKey(index uint32) *ecdsa.PrivateKey {
+	key, ok := chain.keys[index]
+	if ok {
+		return key
+	}
+	key, err := wallet.GetPrvKeyFromMnemonicAndHDWPath(chain.mnemonicPhrase, fmt.Sprintf("m/44'/60'/0'/0/%v", index))
+	if err != nil {
+		panic(err)
+	}
+	chain.keys[index] = key
+	return key
+}
+
+func (chain *Chain) CallOpts(ctx context.Context, index uint32) *bind.CallOpts {
+	opts := chain.TxOpts(ctx, index)
+	return &bind.CallOpts{
+		From:    opts.From,
+		Context: opts.Context,
+	}
+}
+
+func (chain *Chain) LastHeader(ctx context.Context) (*gethtypes.Header, error) {
+	return chain.client.HeaderByNumber(ctx, nil)
+}
+
+func (chain *Chain) GetChannel() core.PathEnd {
+	return chain.Channel
+}
```

Comment:
`Channel` is a public field, so the getter is unnecessary.
Also renaming `Channel` to `PathEnd` makes the code more readable.

Reply:
https://github.com/datachainlab/yui-relayer-build/pull/5/commits/4015b1035b5633a00351baf10ba2dad87f0586ca
rename Channel to PathEnd